pkg/resources/management.cattle.io/v3/cluster/validator.go

```diff
@@ -78,26 +85,36 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return response, nil
 	}
 
-	if request.Operation == admissionv1.Create || request.Operation == admissionv1.Update {
-		cluster, err := objectsv3.ClusterFromRequest(&request.AdmissionRequest)
+	cluster, err := objectsv3.ClusterFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster from request: %w", err)
+	}
+
+	if request.Operation == admissionv1.Create && cluster.Name != "local" {
+		fieldErr, err := common.CheckCreatorPrincipalName(a.userCache, cluster)
```

Comment:
why do we validate this just on create? shouldn't we also check on updates?

Reply:
Under what circumstances we expect the owner of the cluster/project to change? Shouldn't it be immutable? I mean it only make sense when we create a cluster/project and even then if you already have necessary permissions you don't even necessarily need owner RBAC objects. There is even an idea to add an annotation to skip the creation of owner RBAC.
Having said that, we can make sure that both creator id and principal annotations are in fact immutable on update. Though it might be considered a kind of breaking change as we didn't check for that before.

Reply:
> we can make sure that both creator id and principal annotations are in fact immutable on update

I like this idea. It'll stop working if the creator id or principal annotations are manually changed at the moment. Making them immutable makes sense to me.

Reply:
Just make sure that you can change from no annotation -> annotation, just not annotation -> new annotation. Otherwise it'll block the controller that adds the annotation.

Reply:
> Just make sure that you can change from no annotation -> annotation, just not annotation -> new annotation.

We check for immutability only on update [145d37a](https://github.com/rancher/webhook/pull/501/commits/145d37a76aad5cda7e90ab8094f271c5a1c2a843) and the only allowed change to these annotations is to remove them.
These annotations make sense only on create and should be present on the object. Meaning you can't create a cluster and then set these annotations with update.